flux-middle/src/ty/mod.rs

```diff
@@ -88,6 +88,7 @@ pub enum TyKind {
     Uninit,
     Ptr(Path),
     Ref(RefKind, Ty),
+    Constr(Pred, Ty),
```

Comment:
Maybe let's make this an expression. In theory, we should allow kvars here, but fixpoint makes it really hard to have a kvar if it's not directly associated with a binder and we probably shouldn't fight against it right now. I guess there's a practical difference between a normal (existential) refinement type and the composition of an existential + a constr

```suggestion
Constr(Expr, Ty)
```

Reply:
Yes this is easy enough -- went back and forth on this a bit, let me make it `Expr`.